Permanent/ViewControllers/TagDetailsViewController.swift

```diff
@@ -0,0 +1,260 @@
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(doneButtonPressed(_:)))
+        
+        tagFindSearchBar.setDefaultStyle(placeholder: "Add new tag".localized())
+        tagFindSearchBar.setFont(Text.style2.font)
+        tagFindSearchBar.setPlaceholderTextColor(.lightGray)
+        tagFindSearchBar.setTextColor(.lightGray)
+        tagFindSearchBar.setBackgroundColor(.darkGray)
+        tagFindSearchBar.tintColor = .lightGray
+        tagFindSearchBar.barTintColor = .lightGray
+        tagFindSearchBar.autocapitalizationType = .none
+        
+        AddTagButton.configureActionButtonUI(title: "Add".localized(), bgColor: .barneyPurple, buttonHeight: CGFloat(30))
+        
+        tagsCollectionView.backgroundColor = .clear
+        tagsCollectionView.indicatorStyle = .white
+        
+        initTagsState()
+    
+        let columnLayout = CustomViewFlowLayout()
+        tagsCollectionView.collectionViewLayout = columnLayout
+    }
+    
+    @objc func cancelButtonPressed(_ sender: UIBarButtonItem) {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @objc func doneButtonPressed(_ sender: UIBarButtonItem) {
+        let dispatchGroup = DispatchGroup()
+        showSpinner(colored: .lightGray)
+        
+        let forAddingBool = sortedArray.map({ (SortedTagVO) in
+            return SortedTagVO.forAdding
+         })
+        let forAddingNames = sortedArray.map { (item) -> String in
+            return item.tagVO.tagVO.name ?? ""
+        }
+        
+        let addedNames: [String] = zip(forAddingBool, forAddingNames).filter{ $0.0 }.map{ $1 }
+        
+        dispatchGroup.enter()
+        viewModel?.addTag(tagNames: addedNames, completion: { (result) in
+            dispatchGroup.leave()
+        })
+        
+        let forRemovalBool = sortedArray.map { (item) -> Bool in
+            return item.forRemoval
+        }
+        let forRemovalTags = sortedArray.map { (item) -> TagVO in
+            return item.tagVO
+        }
+        let removedTags: [TagVO] = zip(forRemovalBool, forRemovalTags).filter{ $0.0 }.map{ $1 }
+        
+        dispatchGroup.enter()
+        viewModel?.deleteTag(tagVO: removedTags, completion: { (result) in
+            dispatchGroup.leave()
+            })   
+       
+        dispatchGroup.notify(queue: .main) {
+            self.hideSpinner()
+            self.delegate?.tagDetailsViewControllerDidUpdate(self)
+            self.dismiss(animated: true, completion: nil)
+        }
+
+    }
+    
+    @IBAction func AddTagButtonAction(_ sender: Any) {
+        var tagsList: [String] = []
+        
+        for item in sortedArray {
+            if let itemName = item.tagVO.tagVO.name {
+                tagsList.append(itemName)
+            }
+        }
+        
+        if let findTag = tagFindSearchBar.text,
+           !findTag.isEmpty,
+           !tagsList.contains(findTag)
```

Comment:
we can use:
sortedArray.first(where: { $0.tagVO.tagVO.name == findTag }) == nil
instead of the whole tagsList logic.

Reply:
removed tagsList logic.